src/main/java/Controller/CreateMessage.java

```diff
@@ -0,0 +1,121 @@
+package Controller;
+
+import Helper.Helper;
+import Entity.Chat;
+import Entity.Message;
+import DBAccesser.User.QueryUser;
+import DBAccesser.Chat.QueryChat;
+import DBAccesser.Message.QueryMessage;
+import DBAccesser.UserChat.QueryUserChat;
+import DBAccesser.Message.InsertMessage;
+import DBAccesser.Chat.InsertChat;
+import Helper.SuccessResponseGenerator;
+import Exceptions.UserIDDoesNotExistException;
+import Exceptions.ChatIDDoesNotExistException;
+import Exceptions.MessageIDDoesNotExistException;
+import Exceptions.UserChatIDDoesNotExistException;
+import Exceptions.UserIDMissingFromRequestURLPathException;
+import Exceptions.ChatIDMissingFromRequestURLPathException;
+import Exceptions.ContentTypeMissingFromRequestBodyException;
+import Exceptions.TextContentMissingFromRequestBodyException;
+
+import java.util.Map;
+import java.util.LinkedHashMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@RestController
+public class CreateMessage {
+
+    @Autowired
+    QueryUser queryUserHelper;
+
+    @Autowired
+    QueryChat queryChatHelper;
+
+    @Autowired
+    QueryMessage queryMessageHelper;
+
+    @Autowired
+    QueryUserChat queryUserChatHelper;
+
+    @Autowired
+    InsertMessage insertMessageHelper;
+
+    @Autowired
+    InsertChat insertChatHelper;
+
+    @Autowired
+    Helper helper; 
+
+    @PostMapping("/users/chats/messages")
+    public void createMessageWithoutUserIDChatIDPathVariable() {
+
+        String path = "/users/chats/messages";
+
+        throw new UserIDMissingFromRequestURLPathException(path);
+    }
+
+    @PostMapping("/users/chats/{chatID}/messages")
+    public void createMessageWithoutUserIDPathVariable(@PathVariable("chatID") String chatIDString) {
+
+        String path = "/users/chats/" + chatIDString + "/messages";
+
+        throw new UserIDMissingFromRequestURLPathException(path);
+    }
+
+    @PostMapping("/users/{userID}/chats/messages")
+    public void createMessageWithoutChatIDPathVariable(@PathVariable("userID") String userIDString) {
+
+        String path = "/users/" + userIDString + "/chats/messages";
+
+        throw new ChatIDMissingFromRequestURLPathException(path);
+    }
```

Comment:
Have handled them separately so that I can return a custom message for the error that was caught. E.g. if I let it return error automatically, the response would be of the form: 
```
{
    "timestamp": "2020-06-08T15:16:49.600595",
    "status": 400,
    "error": "BAD_REQUEST",
    "message": "",
    "path": "/users/"
}

```
whereas, now it is:

```
{
    "timestamp": "2020-06-08T15:16:49.600595",
    "status": 400,
    "error": "400 BAD_REQUEST",
    "message": "Variable: 'userID' Missing From Request URL Path",
    "path": "/users/"
}
```
This helps me tell where exactly which parameter is missing - making it easier for the client.